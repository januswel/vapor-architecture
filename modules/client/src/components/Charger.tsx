import * as React from 'react'
import { StyleSheet, Text, View } from 'react-native'

import Coin, { Actions as CoinActions } from './Coin'

const styles = StyleSheet.create({
  label: {
    fontSize: 32,
    backgroundColor: 'black',
    color: 'orange',
    textAlign: 'right',
    margin: 8,
    padding: 16,
  },
  coins: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
})

export type ChargedMoney = number
export type Actions = CoinActions
export interface Props {
  chargedMoney: ChargedMoney
  actions: Actions
}

export default (props: Props) => (
  <>
    <Text style={styles.label}>Â¥ {props.chargedMoney}</Text>
    <View style={styles.coins}>
      <Coin {...props} price={10} />
      <Coin {...props} price={50} />
      <Coin {...props} price={100} />
      <Coin {...props} price={500} />
    </View>
  </>
)
