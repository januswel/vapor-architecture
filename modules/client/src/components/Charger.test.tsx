import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Charger from './Charger'

describe('Charger', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Charger chargedMoney={500} actions={{ charge: () => {} }} />)
    expect(component).toMatchSnapshot()
  })
})
