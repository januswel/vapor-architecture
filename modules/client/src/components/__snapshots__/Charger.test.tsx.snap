// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charger renders correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "backgroundColor": "black",
        "color": "orange",
        "fontSize": 32,
        "margin": 8,
        "padding": 16,
        "textAlign": "right",
      }
    }
  >
    Â¥ 
    500
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          10
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          50
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          100
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          500
        </Text>
      </View>
    </View>
  </View>,
]
`;
